pkg/maps/lbmap/lbmap.go

```diff
@@ -561,3 +561,96 @@ func DeleteRevNATBPF(id loadbalancer.ServiceID, isIPv4 bool) error {
 	err := DeleteRevNat(revNATK)
 	return err
 }
+
+// DumpServiceMapsToUserspace dumps the contents of both the IPv6 and IPv4
+// service / loadbalancer BPF maps, and converts them to a SVCMap and slice of
+// LBSVC. IPv4 maps may not be dumped depending on if skipIPv4 is enabled. If
+// includeMasterBackend is true, the returned values will also include services
+// which correspond to "master" backend values in the BPF maps. Returns the
+// errors that occurred while dumping the maps.
+func DumpServiceMapsToUserspace(includeMasterBackend, skipIPv4 bool) (loadbalancer.SVCMap, []*loadbalancer.LBSVC, []error) {
+
+	newSVCMap := loadbalancer.SVCMap{}
+	newSVCList := []*loadbalancer.LBSVC{}
+	errors := make([]error, 0, 2)
+
+	parseSVCEntries := func(key bpf.MapKey, value bpf.MapValue) {
+		svcKey := key.(ServiceKey)
+		//It's the frontend service so we don't add this one
+		if svcKey.GetBackend() == 0 && !includeMasterBackend {
+			return
+		}
+		svcValue := value.(ServiceValue)
+
+		scopedLog := log.WithFields(logrus.Fields{
+			logfields.BPFMapKey:   svcKey,
+			logfields.BPFMapValue: svcValue,
+		})
+
+		scopedLog.Debug("parsing service mapping")
+		fe, be, err := serviceKeynValue2FEnBE(svcKey, svcValue)
+		if err != nil {
+			scopedLog.WithError(err).Error("error converting service key to userspace representation")
+			return
+		}
+
+		svc := newSVCMap.AddFEnBE(fe, be, svcKey.GetBackend())
+		newSVCList = append(newSVCList, svc)
+	}
+
+	if !skipIPv4 {
+		err := Service4Map.DumpWithCallback(parseSVCEntries)
+		if err != nil {
+			errors = append(errors, err)
+		}
+	}
+
+	err := Service6Map.DumpWithCallback(parseSVCEntries)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	return newSVCMap, newSVCList, errors
+}
+
+// DumpRevNATMapsToUserspace dumps the contents of both the IPv6 and IPv4 revNAT
+// BPF maps, and stores the contents of said dumps in a RevNATMap. IPv4 maps may
+// not be dumped depending on if skipIPv4 is enabled. Returns the errors that
+// occurred while dumping the maps.
+func DumpRevNATMapsToUserspace(skipIPv4 bool) (loadbalancer.RevNATMap, []error) {
+
+	newRevNATMap := loadbalancer.RevNATMap{}
+	errors := make([]error, 0, 2)
+
+	parseRevNATEntries := func(key bpf.MapKey, value bpf.MapValue) {
+		revNatK := key.(RevNatKey)
+		revNatV := value.(RevNatValue)
+		scopedLog := log.WithFields(logrus.Fields{
+			logfields.BPFMapKey:   revNatK,
+			logfields.BPFMapValue: revNatV,
+		})
+
+		scopedLog.Debug("parsing BPF revNAT mapping")
+		fe, err := revNatValue2L3n4AddrID(revNatK, revNatV)
+		if err != nil {
```

Comment:
Why isn't this error appended to the `errors` slice?

Reply:
Fixed in next upload.